src/threadNames.cpp

```diff
@@ -0,0 +1,52 @@
+#include "threadNames.h"
+#include "profiler.h"
+
+
+jfieldID ThreadNames::_get_eetop(JNIEnv *env) {
+    MutexLocker ml(_cached_eetop_lock);
+    if (_cached_eetop == NULL) {
+        jclass threadClass = env->FindClass("java/lang/Thread");
+        if (threadClass != NULL) {
+            _cached_eetop = env->GetFieldID(threadClass, "eetop", "J");
+        }
+    }
+    return _cached_eetop;
+}
```

Comment:
Looks like overcomplication. `jfieldID` needs to be initialized just once, and it will be valid throughout the JVM lifetime. It is safe to initialize `jfieldID` at the agent attach time or at the profiler start time without extra lock.

Reply:
OK